### PR TITLE
add OS helper

### PR DESCRIPTION
**Pull request by hjanuschka**
adds more OS detections to Helper class

required for: https://github.com/fastlane/fastlane/pull/6871

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 1]
```diff
@@ -48,6 +48,18 @@ def self.ci?
       return false
     end
 
+    def self.unix?
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/helper.rb]: Not sure what this is useful for in this day and age, I think we should remove it.

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 2]
```diff
@@ -48,6 +48,18 @@ def self.ci?
...
+      !windows?
+    end
+
+    def self.windows?
+      (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
+    end
+
+    def self.linux?
+      unix? and !mac?
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/helper.rb]: Let's not define linux by it being a unix while not being a mac, I'm sure some people would get a heart attack reading this 😅 

I just checked, `RUBY_PLATFORM` is `<arch>-linux` on linux. Maybe a check like that would be more appropriate.

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 3]
```diff
@@ -48,6 +48,18 @@ def self.ci?
       return false
     end
 
+    def self.unix?
+      !windows?
+    end
+
+    def self.windows?
+      (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/helper.rb]: I see that this code is taken from an answer in http://stackoverflow.com/questions/170956/how-can-i-find-which-operating-system-my-ruby-program-is-running-on ... Please add a comment stating where it was taken from.

**Review by milch — CHANGES_REQUESTED** [review]

File: fastlane_core/lib/fastlane_core/helper.rb (hunk 2 above)
Comment (hjanuschka) [inline fastlane_core/lib/fastlane_core/helper.rb]: absolutly right about that.

**Comment by hjanuschka** [comment]
@milch feedback addressed.

**Review by milch — APPROVED** [review]

**Comment by milch** [comment]
Thanks for your contribution! 🚀

**Pull request merged**